### PR TITLE
Aws:ssm:maintenancewindowtask Create Handler

### PR DESCRIPTION
**Pull request by feleyu**
*Issue #, if available:*

*Description of changes:*
Add Implementations of all handlers of MWTask resource. Tested with mvn package && mvn clean verify --no-transfer-progress

Will upgrade to rpdk 2.0 once this pr gets reviewed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-ssm-maintenancewindowtask/src/main/java/software/amazon/ssm/maintenancewindowtask/CreateHandler.java [hunk 1]
```diff
@@ -1,27 +1,79 @@
...
-        final AmazonWebServicesClientProxy proxy,
-        final ResourceHandlerRequest<ResourceModel> request,
-        final CallbackContext callbackContext,
-        final Logger logger) {
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
+
+        logger.log(String.format("Processing CreateHandler request %s", request));
```
Comment (nickysincreate) [inline aws-ssm-maintenancewindowtask/src/main/java/software/amazon/ssm/maintenancewindowtask/CreateHandler.java:46]: I think we do have customer sensitive data in the register task request, can you check on that and make sure we don't log it.

Comment (feleyu) [reply aws-ssm-maintenancewindowtask/src/main/java/software/amazon/ssm/maintenancewindowtask/CreateHandler.java:46]: Yes you are right. All the task parameter details are sensitive data based on https://code.amazon.com/packages/MaintenanceWindowCommon/blobs/mainline/--/src/com/amazonaws/maintenancewindow/datastore/target/WindowTargetRecord.java
https://code.amazon.com/packages/MaintenanceWindowCommon/blobs/mainline/--/src/com/amazonaws/maintenancewindow/datastore/task/WindowTaskRecord.java. I'm working on a string converter right now to log only the insensitive data and I was planning to submit that in the next pr. Would it be better if I submit that in this pr instead?

File: aws-ssm-maintenancewindowtask/src/test/java/software/amazon/ssm/maintenancewindowtask/CreateHandlerTest.java [hunk 2]
```diff
@@ -1,54 +1,209 @@
...
 import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.ssm.maintenancewindowtask.translator.ExceptionTranslator;
+import software.amazon.ssm.maintenancewindowtask.translator.request.RegisterTaskWithMaintenanceWindowTranslator;
+
+import java.util.function.Function;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.*;
+import static software.amazon.ssm.maintenancewindowtask.TestConstants.*;
```
Comment (richardycao) [inline aws-ssm-maintenancewindowtask/src/test/java/software/amazon/ssm/maintenancewindowtask/CreateHandlerTest.java:26]: Wildcard (*) imports should be expanded to be only the imports that are needed. Here is how to turn off auto-correcting imports to wildcard imports: https://intellij-support.jetbrains.com/hc/en-us/community/posts/206203659-Turn-off-Wildcard-imports-

Comment (feleyu) [reply aws-ssm-maintenancewindowtask/src/test/java/software/amazon/ssm/maintenancewindowtask/CreateHandlerTest.java]: Updated. Thank you!

File: aws-ssm-maintenancewindowtask/resource-role.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                - "iam:PassRole"
+                - "ssm:DeregisterTaskFromMaintenanceWindow"
+                - "ssm:DescribeMaintenanceWindowTask"
+                - "ssm:DescribeMaintenanceWindowTasks"
+                - "ssm:GetMaintenanceWindow"
```
Comment (zehuawan) [inline aws-ssm-maintenancewindowtask/resource-role.yaml:30]: Where is GetMaintenanceWindow being used

Comment (feleyu) [reply aws-ssm-maintenancewindowtask/resource-role.yaml:30]: When registerMaintenanceWindowTask, we first getMW from the windowStore and eventually emit the task creation events with window.

File: aws-ssm-maintenancewindowtask/src/main/java/software/amazon/ssm/maintenancewindowtask/ListHandler.java [hunk 4]
```diff
@@ -13,18 +13,18 @@
...
-        final CallbackContext callbackContext,
-        final Logger logger) {
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
 
         final List<ResourceModel> models = new ArrayList<>();
 
         // TODO : put your code here
```
Comment (zehuawan) [inline aws-ssm-maintenancewindowtask/src/main/java/software/amazon/ssm/maintenancewindowtask/ListHandler.java:23]: Remove this

**Pull request merged**